app/assets/javascripts/shared/editor_toolbar.js

```diff
@@ -0,0 +1,232 @@
+/* 
+
+To initialize:
+
+new EditorToolbar($element);
+
+Where `$element` is a jQuery object that has a child input (with type = text) or textarea element.
+Note: Multiple $elements can be stacked: new EditorToolbar($element, $anotherElement, $nthElement);
+
+Each of these input and/or textarea elements will have the Editor Toolbar added.
+
+*/
+
+class EditorToolbar {
+  constructor($target) {
+    this.$target = $target;
+    this.init();
+  }
+
+  init() {
+    // find all child input text and textarea elements
+    this.$target.find('input[type="text"], textarea').each(function() {
+      // wrap each element in a div and add toolbar element as sibling
+      $(this).wrap('<div class="editor-field" data-behavior="editor-field"></div>');
+      $(this).parent().append('<div class="editor-toolbar" data-behavior="editor-toolbar"></div>');
+    });
+
+    // create toolbar buttons for each toolbar
+    $('[data-behavior~=editor-toolbar]').each(function() {
+      if ($(this).prev('textarea').length) { // toolbar for textarea elements
```

Comment:
Not a huge fan of these big blocks inside conditionals. Feels like we could split them into option groups or methods that return this content when asked for.